src/confluence/subcommands/validator.py

```diff
@@ -0,0 +1,77 @@
+import pandas as pd
+
+
+class QMDataFrameValidator(object):
+    def __init__(self):
```

Comment:
I really like this model because it allows the user to add their own callbacks; nicely done! Be sure to add a docstring here. Importantly, that docstring should include a description of the expected signature of the callback function.